⚠️ Nightly upstream-dev CI failed ⚠️

[Workflow Run URL](https://github.com/xarray-contrib/xvec/actions/runs/13042747902)
<details><summary>Python 3.10 Test Summary</summary>

```
xvec/tests/test_zonal_stats.py::test_structure[iterate]: TypeError: Cannot interpret '<geopandas.array.GeometryDtype object at 0x7f05241aded0>' as a data type
xvec/tests/test_zonal_stats.py::test_match: TypeError: Cannot interpret '<geopandas.array.GeometryDtype object at 0x7f05241aded0>' as a data type
xvec/tests/test_zonal_stats.py::test_dataset[iterate]: TypeError: Cannot interpret '<geopandas.array.GeometryDtype object at 0x7f05241aded0>' as a data type
xvec/tests/test_zonal_stats.py::test_dataarray[iterate]: TypeError: Cannot interpret '<geopandas.array.GeometryDtype object at 0x7f05241aded0>' as a data type
xvec/tests/test_zonal_stats.py::test_stat[iterate]: TypeError: Cannot interpret '<geopandas.array.GeometryDtype object at 0x7f05241aded0>' as a data type
xvec/tests/test_zonal_stats.py::test_all_touched[iterate]: TypeError: Cannot interpret '<geopandas.array.GeometryDtype object at 0x7f05241aded0>' as a data type
xvec/tests/test_zonal_stats.py::test_n_jobs: TypeError: Cannot interpret '<geopandas.array.GeometryDtype object at 0x7f05241aded0>' as a data type
xvec/tests/test_zonal_stats.py::test_callable[iterate]: TypeError: Cannot interpret '<geopandas.array.GeometryDtype object at 0x7f05241aded0>' as a data type
xvec/tests/test_zonal_stats.py::test_multiple[iterate]: TypeError: Cannot interpret '<geopandas.array.GeometryDtype object at 0x7f05241aded0>' as a data type
```

</details>

This seems to be an error coming from some change to Xarray between 2024.09 -> 2024.10, I had a quick look for changes to indexing.py in xarray but it's not obvious to me where this is coming from. Full traceback below cc @benbovy @keewis  : 

```pytb
xvec/tests/test_zonal_stats.py:318: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
xvec/accessor.py:1102: in zonal_stats
    result = _zonal_stats_iterative(
xvec/zonal.py:194: in _zonal_stats_iterative
    vec_cube = xr.concat(
../../miniforge3/envs/xvec-dev/lib/python3.12/site-packages/xarray/core/concat.py:277: in concat
    return _dataset_concat(
../../miniforge3/envs/xvec-dev/lib/python3.12/site-packages/xarray/core/concat.py:703: in _dataset_concat
    index_vars = index.create_variables({dim_name: dim_var})
../../miniforge3/envs/xvec-dev/lib/python3.12/site-packages/xarray/core/indexes.py:720: in create_variables
    data = PandasIndexingAdapter(self.index, dtype=self.coord_dtype)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <[AttributeError("'PandasIndexingAdapter' object has no attribute '_dtype'") raised in repr()] PandasIndexingAdapter object at 0x12570b6a0>
array = Index([                                                                                                               ...-59.14999999999992 -51.49999999999997, -58.55000000000007 -51.100000000000016)],
      dtype='object', name='geometry')
dtype = <geopandas.array.GeometryDtype object at 0x1254a1fa0>

    def __init__(self, array: pd.Index, dtype: DTypeLike = None):
        from xarray.core.indexes import safe_cast_to_index
    
        self.array = safe_cast_to_index(array)
    
        if dtype is None:
            self._dtype = get_valid_numpy_dtype(array)
        else:
>           self._dtype = np.dtype(dtype)
E           TypeError: Cannot interpret '<geopandas.array.GeometryDtype object at 0x1254a1fa0>' as a data type

../../miniforge3/envs/xvec-dev/lib/python3.12/site-packages/xarray/core/indexing.py:1674: TypeError
```

--- a/ci/310.yaml
+++ b/ci/310.yaml
@@ -9,6 +9,7 @@ dependencies:
   - rioxarray
   - joblib
   - rasterio
+  - exactextract
   - tqdm
   - pyproj
   # testing
@@ -19,7 +20,3 @@ dependencies:
   - geopandas-base
   - geodatasets
   - pyogrio
-  - pip
-  - pip:
-    - exactextract==0.2.0.dev0
-

--- a/ci/311.yaml
+++ b/ci/311.yaml
@@ -9,6 +9,7 @@ dependencies:
   - rioxarray
   - joblib
   - rasterio
+  - exactextract
   - tqdm
   - pyproj
   # testing
@@ -19,7 +20,3 @@ dependencies:
   - geopandas-base
   - geodatasets
   - pyogrio
-  - pip
-  - pip:
-    - exactextract==0.2.0.dev0
-

--- a/ci/312.yaml
+++ b/ci/312.yaml
@@ -9,7 +9,6 @@ dependencies:
   - rioxarray
   - joblib
   - rasterio
-  - exactextract
   - tqdm
   - pyproj
   # testing
@@ -21,4 +20,8 @@ dependencies:
   - geodatasets
   - pyogrio
   - mypy
+  - pip
+  - pip:
+    - exactextract
+
 

--- a/ci/312.yaml
+++ b/ci/312.yaml
@@ -9,6 +9,7 @@ dependencies:
   - rioxarray
   - joblib
   - rasterio
+  - exactextract
   - tqdm
   - pyproj
   # testing
@@ -20,7 +21,4 @@ dependencies:
   - geodatasets
   - pyogrio
   - mypy
-  - pip
-  - pip:
-    - exactextract==0.2.0.dev0
 

--- a/ci/39.yaml
+++ b/ci/39.yaml
@@ -9,6 +9,7 @@ dependencies:
   - rioxarray
   - joblib
   - rasterio
+  - exactextract
   - tqdm
   - pyproj
   # testing
@@ -19,6 +20,3 @@ dependencies:
   - geopandas-base
   - geodatasets
   - pyogrio
-  - pip
-  - pip:
-    - exactextract==0.2.0.dev0

--- a/ci/dev.yaml
+++ b/ci/dev.yaml
@@ -10,6 +10,7 @@ dependencies:
   - rioxarray
   - joblib
   - rasterio
+  - exactextract
   - tqdm
   # testing
   - pytest
@@ -24,4 +25,3 @@ dependencies:
       - git+https://github.com/shapely/shapely.git@main
       - git+https://github.com/pydata/xarray.git@main
       - git+https://github.com/pyproj4/pyproj.git
-      - exactextract==0.2.0.dev0

--- a/readthedocs.yml
+++ b/readthedocs.yml
@@ -1,5 +1,9 @@
 version: 2
 
+sphinx:
+  # Path to your Sphinx configuration file.
+  configuration: doc/source/conf.py
+
 build:
   os: "ubuntu-20.04"
   tools:

--- a/xvec/zonal.py
+++ b/xvec/zonal.py
@@ -193,7 +193,8 @@ def _zonal_stats_iterative(
         crs = None
     vec_cube = xr.concat(
         zonal,  # type: ignore
-        dim=xr.DataArray(geometry, name=name, dims=name),
+        # astype('O') is a temporary fix for #87
+        dim=xr.DataArray(geometry.astype("O"), name=name, dims=name),
     ).xvec.set_geom_indexes(name, crs=crs)
     gc.collect()
 

--- a/xvec/zonal.py
+++ b/xvec/zonal.py
@@ -194,7 +194,7 @@ def _zonal_stats_iterative(
     vec_cube = xr.concat(
         zonal,  # type: ignore
         # astype('O') is a temporary fix for #87
-        dim=xr.DataArray(geometry.astype("O"), name=name, dims=name),
+        dim=xr.DataArray(np.asarray(geometry).astype("O"), name=name, dims=name),
     ).xvec.set_geom_indexes(name, crs=crs)
     gc.collect()
 

